File: 1052-grumpy-bookstore-owner/1052-grumpy-bookstore-owner.cpp
class Solution {
public:
    int maxSatisfied(vector<int>& customers, vector<int>& grumpy, int minutes) {
        int numAngry = 0;
        int numSat = 0;
        int globalAngry = 0;
        
        for( int i = 0; i < customers.size(); i++) {
            if( grumpy[i] == 0)
                numSat += customers[i];
        }
        
        for( int i = 0; i < minutes; i++) {
            if ( grumpy[i] == 1)
                numAngry += customers[i];
        }
        
        globalAngry = numAngry;
        
        for( int i = minutes; i < customers.size(); i++) {
            if (grumpy[i] == 1)
                numAngry += customers[i];
            if (grumpy[i - minutes] == 1)
                numAngry -= customers[i - minutes];
            globalAngry = max(globalAngry, numAngry);
        }
        
        return (numSat + globalAngry);
    }
};

File: 1052-grumpy-bookstore-owner/1052-grumpy-bookstore-owner.py
class Solution(object):
    def maxSatisfied(self, customers, grumpy, minutes):
        """
        :type customers: List[int]
        :type grumpy: List[int]
        :type minutes: int
        :rtype: int
        """
 
        numsatisfied = 0
        for i in range(len(customers)):
            if grumpy[i] == 0:
                numsatisfied += customers[i]
        
        numangry  = 0
        for i in range(minutes):
            if grumpy[i] == 1:
                numangry += customers[i]
                

        globalmaxangry = numangry
        for i in range(minutes, len(customers)):
            if grumpy[i] == 1:
                numangry += customers[i]
            if grumpy[i-minutes] == 1:
                numangry -= customers[i-minutes]
            globalmaxangry = max(globalmaxangry, numangry)
            
            
        return numsatisfied + globalmaxangry